### DISCUSSION TITLE
Esp32 rechargeable MP3 player with screen

### DISCUSSION THREAD

**Discussion by luke2023** [pschatzmann/arduino-audio-tools#294] [Show and tell]
Hello, I am making a esp32 audio player and  with sd card that can store picture and play music through Bluetooth,it’s like a iPod ,but open source,I am now able to play certain music ,stop, restart,but I wanna start from a specific point like 1:32,any suggestions?(based on player-sd-a2dp example)

**Comment by pschatzmann** [comment]
Create a subclass of AudioStreamX with a AudioStream (which takes the final output) as parameter and the number of seconds to be ignored in the constructor.
From this parameter you can calculate the number of bytes to be ignored (with the help of the number of channels, sample rate and bits_per_sample).
Implement the write() method which ignores the data until the calculated limit is reached and after the limit has been reached you can forward the write to the final output. 

Then you can just add this new class into your output chain. This approach works also with encoded (e.g mp3) data.

**Reply by luke2023** [reply]
@pschatzmann Thank you, I'll try!
